app/models/user.rb

```diff
@@ -19,16 +19,15 @@ class User < ApplicationRecord
   # Restrict the character set for usernames to letters, numbers, and - . _
   validates :username, format: ANCHORED_USERNAME_REGEX
 
+  # Configure default count-per-page for pagination
+  paginates_per 20
+
   # Virtual attribute to allow authenticating by either username or email
   attr_writer :login
   def login
     @login || username || email
   end
 
-  def to_param
```

Comment:
Regarding this commit: 

> Unless I can have to_param work differently in main site and admin area, it needs to go to id 
https://github.com/denny/ShinyCMS-ruby/pull/85/commits/74a4c594bbb7d9e30e82f5a59548c7e7ff27da2e

Consider making use of `CurrentAttributes` to make this possible using request-thread-scoped variables. These docs are well worth a read for an overview: https://api.rubyonrails.org/classes/ActiveSupport/CurrentAttributes.html

A sketch of how this could work:

```rb
class User ...
  
  def to_param
    if Current.area == :admin
      id
    else
      username
    end
  end
```

In `app/controllers/application_controller.rb`, set the current area to something like public/main/non-admin in a before action:

```rb
class ApplicationController ...
  before_action :set_current_area
  
  def set_current_area
    Current.area = :public
  end
```

In `app/controllers/admin_controller.rb`, override `set_current_area`.

```rb
class AdminController ...

  def set_current_area
    Current.area = :admin
  end
```

Create `app/models/current.rb` to make this all work:

```rb
class Current < ActiveSupport::CurrentAttributes
  attribute :area
end
```